pkg/probe/util.go

```diff
@@ -55,3 +57,17 @@ func findPortByName(container *v1.Container, portName string) (int, error) {
 	}
 	return 0, fmt.Errorf("port %s not found", portName)
 }
+
+// ProbeDialer returns a dialer optimized for probes to avoid lingering sockets on TIME-WAIT state.
+// The dialer reduces the TIME-WAIT period to 1 seconds instead of the OS default of 60 seconds.
+// Ref: https://issues.k8s.io/89898
+func ProbeDialer() *net.Dialer {
+	dialer := &net.Dialer{
+		Control: func(network, address string, c syscall.RawConn) error {
+			return c.Control(func(fd uintptr) {
+				syscall.SetsockoptLinger(int(fd), syscall.SOL_SOCKET, syscall.SO_LINGER, &syscall.Linger{Onoff: 1, Linger: 1})
```

Comment:
Does this need to be guarded with Linux-only build tag?  I.e. does it work on Windows?

Reply:
yes, it does, it also fails in arm64 🙃 , I just saw it in the CI jobs

Reply:
fixed